services/fluxmonitor.go

```diff
@@ -390,8 +390,10 @@ func (fm *FluxMonitor) poll() error {
 	}
 
 	median := calculateMedian(values)
+	fm.pollMutex.Lock()
 	fm.latestResult = median
 	fm.latestResultTimestamp = time.Now()
+	fm.pollMutex.Unlock()
```

Comment:
Why do you think so? I put it here mostly for the race condition on this variable. Do you see any other reasons?

Reply:
One reason is that slow adapters could cause this to write an old result here, but we should also avoid polling within the polling interval regardless. Right now it would allow any amount of polling to happen at the same time. Ideally it should only be once within polling interval.

Reply:
True